bson/src/main/org/bson/codecs/BsonCodecProvider.java

```diff
@@ -30,7 +31,9 @@
     @Override
     @SuppressWarnings("unchecked")
     public <T> Codec<T> get(final Class<T> clazz, final CodecRegistry registry) {
-        if (Bson.class.isAssignableFrom(clazz)) {
```

Comment:
This change became necessary after making JsonObject implement Bson (because now the BsonCodecProvider thinks its BsonCodec can encode/decode JsonObjects, but that's not what we want. 

This change also allowed me to delete the JsonObjectCodecProvider because BsonCodecProvider is already part of the default registry. I also added the test in MongoCollectionTest.java to ensure the default registry was correctly encoding/decoding JsonObjects.

Reply:
You can fix that problem by putting the JsonObjectCodecProvider before the BsonCodecProvider in the list of providers.  That's the expected pattern when defining a registry, and there are plenty of cases already where more than one provider will return a codec for a given class (it's why PojoCodecProvider has to go last).

So while this is convenient, I think it's unexpected to include JsonObject here just because it implements Bson interface.

Reply:
Okay that makes sense. Done